content/20_CI_CD_Pipeline/_index.md

```diff
@@ -1,25 +1,81 @@
 
-In the following sections, we will introduce you to AWS CodeBuild and AWS CodePipeline along with setting up the different AWS resources we will be using to use our pipeline.
+AWS CodePipeline can integrate with other services. These can be AWS services such as Amazon Simple Storage Service (Amazon S3), or third-party products such as GitHub, Docker BuildKit, Docker Scout and Docker Hub.
+
+![Docker](/images/cicdpipeline2.png)
+
+
+### AWS CI/CD pipeline components
+AWS can set up CI/CD pipelines using the following AWS developer tools:
+
+    - AWS CodeConnections (to use GitHub repository).
+    - AWS CodeBuild.
+    - AWS CodePipeline.
+    - AWS CodeDeploy.
+    - Amazon Elastic Container Registry.
+    - Docker Build Cloud
+    - Docker Scout
+    - Docker TestContainers
+    - Docker Hub
+
+The creation of CI/CD pipelines using these services can be automated using AWS CDK and AWS CloudFormation.
+ 
+
+### Continuous Integration 
+
+Continuous Integration (CI) is a DevOps software development practice where developers regularly merge their code changes into a central repository, after which automated builds and tests are run. Continuous Integration most often refers to the build or integration stage of the software release process and entails both an automation component (e.g. a CI or build service) and a cultural component (e.g. learning to integrate frequently). The key goals of Continuous Integration are to find and address bugs quicker, improve software quality, and reduce the time it takes to validate and release new software updates.
+
+### Why is Continuous Integration needed?
+
+In the past, developers on a team might work in isolation for an extended period of time and only merge their changes to the mainline branch once their work was completed. This made merging code changes difficult and time-consuming, and also resulted in bugs accumulating for a long time without correction. These factors made it harder to deliver updates to customers quickly.
+
+
+### How does Continuous Integration work?
+
+With Continuous Integration, developers frequently commit to a shared repository using a version control system such as Git. Prior to each commit, developers may choose to run local unit tests on their code as an extra verification layer before integrating. A Continuous Integration service automatically builds and runs unit tests on the new code changes to immediately surface any errors.
+
+![Docker](/images/cicdpipeline3.png)
```

Comment:
could we add a better substitued name so instead of Docker we can use `![cicdpipeline3](/images/cicdpipeline3.png)`